Polyhedron/demo/Polyhedron/MainWindow.cpp

```diff
@@ -1945,3 +1928,53 @@ void MainWindow::resetHeader()
 }
 
 
+void MainWindow::reset_default_loaders()
+{
+  default_plugin_selection.clear();
+
+  const char* prop_name = "Menu modified by MainWindow.";
+  QMenu* menu = ui->menuFile;
+  if(!menu)
+    return;
+  bool menuChanged = menu->property(prop_name).toBool();
+  if(!menuChanged) {
+    menu->setProperty(prop_name, true);
+  }
+  QList<QAction*> menuActions = menu->actions();
+  menu->removeAction(actionResetDefaultLoaders);
+}
+
+void MainWindow::insertActionAfter(QMenu* menu, QString actionAfterName, QAction* actionToInsert)
+{
+  const char* prop_name = "Menu modified by MainWindow.";
+  if(menu)
+  {
+    bool menuChanged = menu->property(prop_name).toBool();
+    if(!menuChanged) {
+      menu->setProperty(prop_name, true);
+    }
+    QList<QAction*> menuActions = menu->actions();
+    if(menuActions.contains(actionToInsert))
+      return;
+    // Look for action just after "actionAfterName..." action
+    QAction* actionAfter = NULL;
+    for ( QList<QAction*>::iterator it_action = menuActions.begin(),
+          end = menuActions.end() ; it_action != end ; ++ it_action )
+    {
+      if ( NULL != *it_action && (*it_action)->text().contains(actionAfterName) )
+      {
+        ++it_action;
+        if ( it_action != end && NULL != *it_action )
+        {
+          actionAfter = *it_action;
+        }
+      }
+    }
+
+    // Insert "Load implicit function" action
+    if ( NULL != actionAfter )
+    {
+      menu->insertAction(actionAfter,actionToInsert);
+    }
+  }
+}
```

Comment:
That function is complicated, and there is a lot better to do using the Qt API.

The "Load" action is simply `ui->actionLoad`, or (in case you do not have a direct access to the action pointer) `this->findChild<QAction*>("actionLoad")`.

Then you can do:
```C++
if(!menu->actions().contains(actionToInsert)) {
  menu>insertAction(ui->actionLoad, actionToInsert);
}
```